src/w3d/lib/random.cpp

```diff
@@ -158,7 +158,7 @@ int Random4Class::operator()()
     y ^= TEMPERING_SHIFT_T(y) & TEMPERING_MASK_C;
     y ^= TEMPERING_SHIFT_L(y);
 
-    int *x = (int *)&y;
+    int *x = reinterpret_cast<int *>(&y);
```

Comment:
ummm what is this function doing?? It can just `return static_cast<int32_t>(y);` and skip the x completely.

Reply:
Actually that would not be the same. `static_cast` would perform conversion. Currently it does not and simply takes unsigned value and treats it as is for int. Changing this would change logic, which I would like to not do here without understanding the consequences.

Reply:
That still doesn't make sense if that was the case you would reinterpret cast straight to the unsigned value, no need for the pointers.

https://stackoverflow.com/questions/1751346/interpret-signed-as-unsigned

It looks like this is changing defined behaviour for implementation defined behaviour

Reply:
I can tell you right now that
`int *x = reinterpret_cast<int *>(&y)`
does not yield same result as
`int x = static_cast(y)`

The first will create type without conversion and the second with conversion. I will not change this for the sake of not altering this algorithm.

Reply:
https://godbolt.org/z/cjEzMoGxa its legal but its pointless. `static_cast` will always produce the same result. You can see it in the attached all the mov's are exactly the same except the reinterpret that has to add an extra load.

Reply:
Alright then. I changed to static casts. I am still not convinced that this is 100% identical behavior, but if you vouch for that, then you can Approve it like so.